sources/api/pluto/src/ec2.rs

```diff
@@ -51,26 +57,33 @@ pub(super) async fn get_private_dns_name(region: &str, instance_id: &str) -> Res
     };
 
     tokio::time::timeout(
-        EC2_DESCRIBE_INSTANCES_TIMEOUT,
-        client
-            .describe_instances()
-            .instance_ids(instance_id.to_owned())
-            .send(),
+        FETCH_PRIVATE_DNS_NAME_TIMEOUT,
+        Retry::spawn(
+            FibonacciBackoff::from_millis(FIBONACCI_BACKOFF_BASE_DURATION_MILLIS).map(jitter),
+            || async {
+                client
+                    .describe_instances()
+                    .instance_ids(instance_id.to_owned())
+                    .send()
+                    .await
+                    .context(DescribeInstancesSnafu { instance_id })?
+                    .reservations
+                    .and_then(|reservations| {
+                        reservations.first().and_then(|r| {
+                            r.instances.clone().and_then(|instances| {
+                                instances
+                                    .first()
+                                    .and_then(|i| i.private_dns_name().map(|s| s.to_string()))
+                            })
+                        })
+                    })
+                    .filter(|private_dns_name| !private_dns_name.is_empty())
+                    .context(MissingSnafu {
+                        field: "Reservation.Instance.PrivateDNSName",
+                    })
+            },
```

Comment:
If there are valid scenarios where the EC2 PrivateDNSName might be empty and the node can still join a cluster, then I can't really endorse this since all those nodes would now take 5 minutes longer to start on first boot.

Reply:
Can a node actually still join the cluster if this fails? If so, we likely need to re-evaluate our timeout and logic to not sit here for 5 minutes. I assumed this 5 minute timeout was more or less just there to let the cluster realize the node is not going to be healthy enough to connect to the cluster. Either way, 5 minutes still feels too long for this timeout.

Reply:
I don't believe it can still join. The node's boot fails if this fails, so there doesn't seem to be much harm in waiting and giving things a chance to resolve.

I tried joining a node to a cluster in a VPC with DNS disabled. The instance was created and started to boot, but system console ended with:

```
[  312.455248] sundog[983]: Setting generator 'pluto private-dns-name' failed with exit code 1 - stderr: Timed out retrieving private DNS name from EC2: deadline has elapsed
[FAILED] Failed to start User-specified setting generators.
See 'systemctl status sundog.service' for details.
[DEPEND] Dependency failed for Bottlerocket initial configuration complete.
[DEPEND] Dependency failed for Isolates configured.target.
[DEPEND] Dependency failed for Applies settings to create config files.
[DEPEND] Dependency failed for Sets the hostname.
[DEPEND] Dependency failed for Send signal to CloudFormation Stack.
```

The control container never runs, so no chance to get in to the console to explore more. But the node is basically DOA at this point.

Interestingly, the `eksctl create nodegroup` timeout appears to be 25 minutes. So even after this node failed, without specifying a different timeout to the `eksctl` command it waited for ~20 more minutes before failing.

As a test, I provided a `settings.kubernetes.host-name` value in my user data. Since this was a random string, it still failed to join the k8s cluster. But the system log showed that it got past this point in initialization with no delay or other errors.